### DISCUSSION TITLE
AFFiNE 是否支持 OpenLDAP 登录？

### DISCUSSION THREAD

**Discussion by MartinLau7** [toeverything/AFFiNE#1091] [Q&A]
我打算在内网搭建一套，但我没有找到相关 LDAP 的支持，是否未来会加入 OpenLDAP 登录支持？

**Comment by himself65** [comment]
Recently we don't have the plan to support it. But If you want, you can implement a provider yourself.

https://github.com/toeverything/AFFiNE/tree/master/packages/data-center/src/provider

You need to extend a class here

https://github.com/toeverything/AFFiNE/blob/b6906467be752ff602d6cd772288461b19c22b31/packages/data-center/src/provider/base.ts#L25

**Reply by MartinLau7** [reply]
期待后续加入

**Reply by knurdmas** [reply]
> Recently we don't have the plan to support it. But If you want, you can implement a provider yourself.
> 
> https://github.com/toeverything/AFFiNE/tree/master/packages/data-center/src/provider
> 
> You need to extend a class here
> 
> https://github.com/toeverything/AFFiNE/blob/b6906467be752ff602d6cd772288461b19c22b31/packages/data-center/src/provider/base.ts#L25

Hi! Sorry, I can't find base.ts. What should i do to try creating LDAP auth now?

**Reply by MartinLau7** [reply]
> > Recently we don't have the plan to support it. But If you want, you can implement a provider yourself.
> > https://github.com/toeverything/AFFiNE/tree/master/packages/data-center/src/provider
> > You need to extend a class here
> > https://github.com/toeverything/AFFiNE/blob/b6906467be752ff602d6cd772288461b19c22b31/packages/data-center/src/provider/base.ts#L25
> 
> Hi! Sorry, I can't find base.ts. What should i do to try creating LDAP auth now?

That part is missing from the latest code. @Himself65
